### DISCUSSION TITLE
how to modify dag name and keep history tasks

### DISCUSSION THREAD

**Discussion by fb-hdm** [apache/airflow#13113] [General] [answered]
Hello,
I need to modify a dag name for an existing dag and I want to keep all my history tasks. My airflow version is 1.10.10
I did not find any cli command or api to do that and my option would be to update manually the sql.

First question is, is the right solution doing through sql?
Second question is, if yes do I have to update just the column dag_id when the col is present in a table?

Thank you in advance
Regards
Fb

**Comment by turbaszek** [comment]
@fb-hdm what do you exactly mean by
> want to keep all my history tasks

What would you like to achieve?

**Comment by fb-hdm** [comment]
@turbaszek 
thank you for your answer.
I have already a dag with many executions (with many task executed), what I want to achieve is:
- rename the dag
- maintain all the past executions even after I will rename the dag

Hope now is more clear
Thank you again
fb

**Comment by turbaszek** [comment]
This should work out of the box. If you rename the DAG, Airflow will create a new one with the new name and old metadata will be there. However, if you want to remove the DAG from webui this will remove all metadata.

**Comment by fb-hdm** [comment, marked as answer]
That's true, but I just want to rename the dag and not create e new one. Is it possible?

**Reply by turbaszek** [reply]
> That's true, but I just want to rename the dag and not create e new one. Is it possible?

You would need to update the `dag_id` in every table in airflow metadatabase for this particular DAG. I don't see any other solution.

**Reply by fb-hdm** [reply]
@turbaszek then I'm doing via sql. thank you for your help

**Reply by turbaszek** [reply]
@fb-hdm could you please mark the question as answered?

**Comment by abbas-taqvi** [comment]
Hey @turbaszek, were you able to make a sql command for this?
